text/0000-stable-simd.md

```diff
@@ -0,0 +1,720 @@
+  however, will be convertible between one another (either via transmutes or via
+  explicit functions). This conversion will have zero run-time cost.
+* The portable simd types will likely live in a module like `std::simd` rather
+  than `std::vendor`.
+
+The design around these portable types are ongoing, however, and stay tuned for
+an RFC for the `std::simd` module!
+
+[simd.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SIMD
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Stable SIMD in Rust ends up requiring a surprising number of both language and
+library features to be productive. Thankfully, though, there's has been quite a
+bit of experimentation over time with SIMD in Rust and we've gotten a lot of
+good experience along the way! In this section, though, we'll be going into the
+various features in detail.
+
+## The `#[target_feature]` Attribute
+
+The `#[target_feature]` attribute was specified in [RFC 2045][rfc2045] and
+remains unchanged from that specification. As a quick recap it allows you to add
+this attribute to functions:
+
+```rust
+#[target_feature(enable = "avx2")]
+```
+
+The only currently allowed key is `enable` (one day we may allow `disable`). The
+string values accepted by `enable` will be separately stabilized but are likely
+to be guided by vendor definitions. For example in Intel's [intrinsic
+guide][intel-intr] it lists functions under "AVX2", so we're likely to stabilize
+the name `avx2` for Rust.
+
+There's a good number of these features supported by the compiler today. It's
+expected that when stabilizing other pieces of this RFC the names of the
+following existing features for x86 will be stabilized:
+
+* `aes`
+* `avx2`
+* `avx`
+* `bmi2`
+* `bmi` - to be renamed to `bmi1`, the name Intel gives it
+* `fma`
+* `fxsr`
+* `lzcnt`
+* `popcnt`
+* `rdrnd`
+* `rdseed`
+* `sse2`
+* `sse3`
+* `sse4.1`
+* `sse4.2`
+* `sse`
+* `ssse3`
+* `xsave`
+* `xsavec`
+* `xsaveopt`
+* `xsaves`
+
+Note that AVX-512 names are missing from this list, but that's because we
+haven't implemented any AVX-512 intrinsics yet. Those'll get stabilized on their
+own once implemented. Additionally note that `mmx` is missing from this list.
+For reasons discussed later, it's proposed that MMX types are omitted from the
+first pass of stabilization. AMD also has some specific features supported
+(`sse4a`, `tbm`), and so do ARM, MIPS, and PowerPC, but none of these feature
+names a proposed for becoming stable in the first pass.
+
+## The `target_feature` value in `#[cfg]`
+
+In addition to enabling target features for a function the compiler will also
+allow statically testing whether a particular target feature is enabled. This
+corresponds to the `cfg_target_feature` feature today in rustc, and can be seen
+via:
+
+```rust
+#[cfg(target_feature = "avx)]
+fn foo() {
+    // implementation that can use `avx`
+}
+
+#[cfg(not(target_feature = "avx))]
```

Comment:
missing `"`

Reply:
@alexcrichton - Good to know that I'm actually starting to understand Rust code from reading these RFC's. I really appreciate all the work that those like you put into these RFC's for those of us still learning Rust.